parsing/parser.mly

```diff
@@ -886,12 +886,13 @@ toplevel_phrase:
   | toplevel_directive SEMISEMI          { $1 }
   | EOF                                  { raise End_of_file }
 ;
-top_structure: extra_str(top_structure_nodoc) { $1 }
-top_structure_nodoc:
+%inline top_structure:
+  extra_str(
     seq_expr post_item_attributes
       { text_str $startpos($1) @ [mkstrexp $1 $2] }
   | top_structure_tail_nodoc
       { $1 }
+  ) { $1 }
```

Comment:
Can the same transformation be applied to `use_file/use_file_body` below?

Reply:
@nojb, the answer is not clear to me. `use_file_body` is used in two places, and `use_file_body` and `use_file_tail` are mutually recursive. I could not make sense of these definitions at first sight, so I did not attempt to modify them. The fact that there are side effects in the semantic actions (`mark_rhs_docs`) worries me further.

Other definitions, like `structure_nodoc` and `structure_tail_nodoc`, look as though they should be amenable to simplification (ideally, a structure should be just a list of structure items...) yet it is difficult to make sense of them in their present form.